user-impl/src/test/java/com/example/auction/user/impl/UserRepositoryTest.java

```diff
@@ -0,0 +1,111 @@
+
+public class UserRepositoryTest {
+
+    private final static ServiceTest.Setup setup = defaultSetup().withCassandra(true)
+            .configureBuilder(b ->
+                    // by default, cassandra-query-journal delays propagation of events by 10sec. In test we're using
+                    // a 1 node cluster so this delay is not necessary.
+                    b.configure("cassandra-query-journal.eventual-consistency-delay", "0")
+                            .overrides(bind(ReadSide.class).to(ReadSideTestDriver.class),
+                                    bind(TopicFactory.class).to(DoNothingTopicFactory.class))
+            );
+
+    private static ServiceTest.TestServer testServer;
+
+    @BeforeClass
+    public static void beforeAll() {
+        testServer = ServiceTest.startServer(setup);
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        testServer.stop();
+    }
+
+    private ReadSideTestDriver testDriver = testServer.injector().instanceOf(ReadSideTestDriver.class);
+    private UserRepository userRepository = testServer.injector().instanceOf(UserRepository.class);
+    private AtomicInteger offset;
+
+    private final UUID userId = UUID.randomUUID();
+    private final String name = "admin";
+    private final String email = "admin@gmail.com";
+    private final String password = PUserCommand.hashPassword("admin");
+    private final Instant createdAt = Instant.now();
+    private final PUser userCreated = new PUser(userId, createdAt, name, email, password);
+
+
+    @Before
+    public void restartOffset() {
+        offset = new AtomicInteger(1);
+    }
+
+
+    public PaginatedSequence<User> shouldGetUsers() throws InterruptedException, ExecutionException, TimeoutException {
+        return Await.result(userRepository.getUsers(0, 10));
+    }
+
+    @Test
+    public void shouldCreateUser() throws InterruptedException, ExecutionException, TimeoutException {
+        feed(new PUserEvent.PUserCreated(userCreated));
+        PaginatedSequence<User> users = shouldGetUsers();
+        assertEquals(1, users.getCount());
+        User expected = new User(userId, createdAt, name, email);
+        assertEquals(expected, users.getItems().get(0));
+    }
+
+    @Test
+    public void shouldPaginateUserRetrieval() throws InterruptedException, ExecutionException, TimeoutException {
+
+        PaginatedSequence<User> usersCreatedBefore = Await.result(userRepository.getUsers(1, 10));
+        int initialCount = usersCreatedBefore.getCount();
+        int size = 25;
+
+        for (int i = 0; i < size; i++) {
+            feed(new PUserEvent.PUserCreated(buildFixture(createdAt, i)));
+        }
+
+        PaginatedSequence<User> createdUsers = Await.result(userRepository.getUsers(1, 10));
+        assertEquals(size + initialCount, createdUsers.getCount());
+        assertEquals(10, createdUsers.getItems().size());
+        assertEquals("admin11", createdUsers.getItems().get(4).getName());
```

Comment:
This one will randomly fail because the ordering is not guaranteed. 

`UserRepository.selectUsers` must include a `order by createdAt` to make this call deterministic.